src/library/scala/math/BigInt.scala

```diff
@@ -291,38 +467,61 @@ final class BigInt(val bigInteger: BigInteger)
 
   /** Returns the bitwise complement of this BigInt
    */
-  def unary_~ : BigInt = new BigInt(this.bigInteger.not())
+  def unary_~ : BigInt =
+    // it is equal to -(this + 1)
+    if (longEncoding && _long != Long.MaxValue) BigInt(-(_long + 1)) else new BigInt(this.bigInteger.not())
 
   /** Returns true if and only if the designated bit is set.
    */
-  def testBit (n: Int): Boolean = this.bigInteger.testBit(n)
+  def testBit (n: Int): Boolean =
```

Comment:
are there laws that these functions all behave the same as BigInteger? I think those would be good laws to have.

Reply:
we do have ScalaCheck in this repo, so yeah, I agree some property-based tests should be added

Reply:
can probably also be copied from Spire

Reply:
There is a ScalaCheck property for testBit, setBit, clearBit, flipBit etc.